browser/src/Editor/NeovimEditor/NeovimSurface.tsx

```diff
@@ -67,7 +70,7 @@ class NeovimSurface extends React.Component<INeovimSurfaceProps> {
 
     public render(): JSX.Element {
         return (
-            <div className="container vertical full">
+            <div className="container vertical full" ref={e => (this._editorContainer = e)}>
```

Comment:
What if we made the following changes?

1) Use `<FileDropHandler />` for this line
2) Update `<FileDropHandler>` to render the children (see note above)
3) Have `FileDropHandler` be responsible for getting the ref
4) If `FileDropHandler` owns the ref, it can hook those events on the element, and call `preventDefault` - so now `NeovimEditor` doesn't need to explicitly hook any drag/drop events - it can delegate all the functionality to `FileDropHandler`.

Right now, the `FileDropHandler` feels a bit 'awkward' - it doesn't feel like a first class React component, since it's position in the hierarchy really doesn't matter - all that matters is the target thats passed in. But I think it'd be simpler if we could just put this as the 'container' and have it take care of the events (including the `preventDefaults`).

Reply:
@bryphe by using the handler in place of the container the component becomes entirely specific to this one use i've refactored the global handlers into the component and use the [ref callback pattern](https://reactjs.org/docs/refs-and-the-dom.html#callback-refs) which I pass down using a render prop aka

```js
<FileDropHandler>
	{({ setRef }) => (
		<div ref={setRef}/>
	}
</FileDropHandler>
```

this has the advantage of allowing the file handler to be reused without needing to render a particular component of a given size, an example of a potential use case of this could be having the screen actually have two drop containers (invisible) and on hovering a file onto oni if on the left then that handler could open in a `vsplit` and vice versa, or potentially dropping an image into a markdown file and having the link extracted and input into the document etc.